### DISCUSSION TITLE
Items that used to display in the viewer do not display in the current version.

### DISCUSSION THREAD

**Discussion by photoMike** [google/model-viewer#4297] [Q&A]
### Description

I have an NFT that is displayed on pool.pm which uses model-viewer to display glb files from ifps. I am not sure when it started happening but the wood frame for the painting stopped rendering completely. I have tested the original glb file on the model-viewer editor page and can confirm that it also does not display the frame anymore. It also gives me some errors that I have no idea what to do with. (I'm not a coder). I'll paste them below:

The original Twitter thread between myself and the web dev is here:
https://twitter.com/LightCaptive/status/1660874886976360453 

<html><body>
<div class="report">
  <h1>Validation report</h1>
  <ul>
    <li><b>Format:</b> glTF 2.0</li>
    <li><b>Generator:</b> Khronos glTF Blender I/O v1.6.16</li>
    <li>
      <b>Stats:</b>
      <ul>
        <li>14 draw calls</li>
        <li>0 animations</li>
        <li>0 joints</li>
        <li>11 materials</li>
        <li>42856 vertices</li>
        <li>28583 triangles</li>
        <li>2.00 m x-width</li>
        <li>2.20 m y-height</li>
        <li>0.363 m z-length</li>
        <li>Extensions used: KHR_texture_transform, KHR_lights_punctual</li>
        
      </ul>
    </li>
  </ul>
  <hr>
  <p>
    Report generated by
    <a href="https://github.com/KhronosGroup/glTF-Validator/">KhronosGroup/glTF-Validator</a>
    2.0.0-dev.3.9.
  </p>
</div>
    
    

Error | Message | Pointer
-- | -- | --
MESH_PRIMITIVE_TOO_FEW_TEXCOORDS | Material is incompatible with mesh primitive: Texture binding  '/materials/0/pbrMetallicRoughness/baseColorTexture' needs 'TEXCOORD_1'  attribute. | /meshes/0/primitives/0/material
MESH_PRIMITIVE_TOO_FEW_TEXCOORDS | Material is incompatible with mesh primitive: Texture binding  '/materials/0/pbrMetallicRoughness/metallicRoughnessTexture' needs  'TEXCOORD_1' attribute. | /meshes/0/primitives/0/material
MESH_PRIMITIVE_TOO_FEW_TEXCOORDS | Material is incompatible with mesh primitive: Texture binding '/materials/0/normalTexture' needs 'TEXCOORD_1' attribute. | /meshes/0/primitives/0/material
ACCESSOR_INVALID_FLOAT | Accessor element at index 58 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 59 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 66 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 67 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 68 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 69 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 74 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 75 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 80 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 81 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 82 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 83 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 114 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 115 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 116 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 117 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 124 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 125 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 136 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 137 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 138 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 139 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 148 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 149 is NaN. | /accessors/44
ACCESSOR_INVALID_FLOAT | Accessor element at index 58 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 59 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 66 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 67 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 68 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 69 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 74 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 75 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 80 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 81 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 82 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 83 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 114 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 115 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 116 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 117 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 124 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 125 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 136 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 137 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 138 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 139 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 148 is NaN. | /accessors/45
ACCESSOR_INVALID_FLOAT | Accessor element at index 149 is NaN. | /accessors/45

</body>
</html>Validation report

    Format: glTF 2.0
    Generator: Khronos glTF Blender I/O v1.6.16
    Stats:
        14 draw calls
        0 animations
        0 joints
        11 materials
        42856 vertices
        28583 triangles
        2.00 m x-width
        2.20 m y-height
        0.363 m z-length
        Extensions used: KHR_texture_transform, KHR_lights_punctual

Report generated by [KhronosGroup/glTF-Validator](https://github.com/KhronosGroup/glTF-Validator/) 2.0.0-dev.3.9.
Error 	Message 	Pointer
MESH_PRIMITIVE_TOO_FEW_TEXCOORDS 	Material is incompatible with mesh primitive: Texture binding '/materials/0/pbrMetallicRoughness/baseColorTexture' needs 'TEXCOORD_1' attribute. 	/meshes/0/primitives/0/material
MESH_PRIMITIVE_TOO_FEW_TEXCOORDS 	Material is incompatible with mesh primitive: Texture binding '/materials/0/pbrMetallicRoughness/metallicRoughnessTexture' needs 'TEXCOORD_1' attribute. 	/meshes/0/primitives/0/material
MESH_PRIMITIVE_TOO_FEW_TEXCOORDS 	Material is incompatible with mesh primitive: Texture binding '/materials/0/normalTexture' needs 'TEXCOORD_1' attribute. 	/meshes/0/primitives/0/material
ACCESSOR_INVALID_FLOAT 	Accessor element at index 58 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 59 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 66 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 67 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 68 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 69 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 74 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 75 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 80 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 81 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 82 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 83 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 114 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 115 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 116 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 117 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 124 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 125 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 136 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 137 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 138 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 139 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 148 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 149 is NaN. 	/accessors/44
ACCESSOR_INVALID_FLOAT 	Accessor element at index 58 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 59 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 66 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 67 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 68 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 69 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 74 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 75 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 80 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 81 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 82 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 83 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 114 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 115 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 116 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 117 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 124 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 125 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 136 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 137 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 138 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 139 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 148 is NaN. 	/accessors/45
ACCESSOR_INVALID_FLOAT 	Accessor element at index 149 is NaN. 	/accessors/45
Hint 	Message 	Pointer
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/0/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/0/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/0/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/0/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/1/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/1/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/1/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/1/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/2/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/2/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/2/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/2/primitives/0/attributes/TEXCOORD_1
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/3/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/3/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/3/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/3/primitives/0/attributes/TEXCOORD_1
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/3/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/4/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/4/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/4/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/4/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/5/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/5/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/5/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/5/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/6/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/6/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/6/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/6/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/7/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/7/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/7/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/7/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/8/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/8/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/8/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/9/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/9/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/9/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/10/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/10/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/10/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/11/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/11/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/11/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/11/primitives/0/attributes/TEXCOORD_1
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/11/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/12/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/12/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/12/primitives/0/attributes/TEXCOORD_0
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/12/primitives/0/indices
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/13/primitives/0/attributes/POSITION
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/13/primitives/0/attributes/NORMAL
BUFFER_VIEW_TARGET_MISSING 	bufferView.target should be set for vertex or index data. 	/meshes/13/primitives/0/attributes/TEXCOORD_0
Info 	Message 	Pointer
UNUSED_OBJECT 	This object may be unused. 	/meshes/0/primitives/0/attributes/TEXCOORD_0
UNUSED_OBJECT 	This object may be unused. 	/meshes/2/primitives/0/attributes/TEXCOORD_1
UNUSED_OBJECT 	This object may be unused. 	/meshes/3/primitives/0/attributes/TEXCOORD_1
UNUSED_OBJECT 	This object may be unused. 	/meshes/4/primitives/0/attributes/TEXCOORD_0
UNUSED_OBJECT 	This object may be unused. 	/meshes/11/primitives/0/attributes/TEXCOORD_0
UNUSED_OBJECT 	This object may be unused. 	/meshes/11/primitives/0/attributes/TEXCOORD_1
UNUSED_OBJECT 	This object may be unused. 	/meshes/12/primitives/0/attributes/TEXCOORD_0
UNUSED_OBJECT 	This object may be unused. 	/meshes/13/primitives/0/attributes/TEXCOORD_0
IMAGE_NPOT_DIMENSIONS 	Image has non-power-of-two dimensions: 2160x2160. 	/images/3
IMAGE_NPOT_DIMENSIONS 	Image has non-power-of-two dimensions: 2160x2160. 	/images/4

#### Live Demo
https://pool.pm/asset14lmxmn62qaks8xkyxdl5j0m9h8werx3mvjddee 

### Version
I don't have control of the pool.pm script. The issue presents on the current model-viewer editor though. 

### Browser Affected

- [x ] Chrome, version: 113.0.5672.127
- [ ] Edge
- [x ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [x ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [x ] Windows

**Comment by elalish** [comment]
Those errors mean it's an invalid GLB, which means it only worked before due to luck. You'll need to go back to whatever software exported the GLB and re-export, hoping they've fixed their bugs in the meanwhile. If that still doesn't work, you should file a bug on their software. Always check your GLBs in the validator, especially before selling them to anyone!

**Reply by photoMike** [reply]
Thanks for the reply, The software I created the glb file with was Blender 2.9 I believe, (Generator: Khronos glTF Blender I/O v1.6.16) and I used your model-viewer editor to check the model before uploading it to IFPS on 2021-09-18. No errors at that time. I am unable to change the file because it's been uploaded to IFPS and the NFT can't reference a new file as the policy is locked. Something has changed with model-viewer in the last few months. Everything was absolutely fine before.
